internal/pkg/aws/s3/s3.go

```diff
@@ -101,10 +82,19 @@ func (s *S3) ZipAndUpload(bucket, key string, files ...NamedBinary) (string, err
 }
 
 // Upload uploads a file to an S3 bucket under the specified key.
-func (s *S3) Upload(bucket, key string, file NamedBinary) (string, error) {
-	buf := new(bytes.Buffer)
-	buf.Write(file.Content())
-	return s.upload(bucket, key, buf)
+func (s *S3) Upload(bucket, key string, data io.Reader) (string, error) {
+	return s.upload(bucket, key, data)
+}
+
+// MkdirTimestamp prefixes the key with the current timestamp "manual/<timestamp>/key".
+func MkdirTimestamp(key string) string {
+	id := time.Now().Unix()
+	return path.Join(artifactDirName, strconv.FormatInt(id, 10), key)
+}
+
+// MkdirSHA prefixes the key with the SHA256 hash of the contents of r "<hash>/key".
```

Comment:
nit:
```suggestion
// MkdirSHA prefixes the key with the SHA256 hash of the contents of r "manual/<hash>/key".
```

Reply:
I guess that means the value of <hash> will get replaced? I'll ditch for consistency